Tweak untestable conformance language

garethsb opened the pull request.

garethsb committed:
Tweak untestable conformance language

N-Nagorny commented on docs/Controllers.md:
```diff
@@ -82,7 +82,7 @@ In order to avoid polling of the HTTP API, it is RECOMMENDED that Controllers us
 ## Dynamic Update of Resources
 
-The Controller MUST be capable of using the Query API to discover and dynamically report the state of any registered resource, including Node, Device, Source, Flow, Sender, and Receiver.
+The Controller uses the Query API to discover and track the state of any registered resource, including Node, Device, Source, Flow, Sender, and Receiver.
```
Should we mention that there is no Query API in the Peer to Peer mode and give some guidance on what to do in this case?

garethsb replied:
Pretty much everything else in this Controller spec assumes use of the Query API. I don't disagree that advice about what to do in the absence of a Registry is a good idea, but that's a separate topic/PR it seems to me.

garethsb replied:
On second thoughts, the possibility of operating in peer-to-peer mode could be why the spec was written that way, although no-one remembered that when I asked. 🙂

I realise there's another instance of "The Controller MUST be capable of..." a bit earlier in the document too.

Perhaps separating it into two sentences, i.e. "The Controller MUST be capable of operating in registered mode." followed by adding "In registered operation,..." at the start of proposed change is a better idea?

N-Nagorny replied:
@garethsb, yes, should be very clear for readers.

garethsb replied:
Updated, ready for review again.

garethsb committed:
Distinguish between registered and peer-to-peer operation

garethsb committed:
The Controller MUST support registered operation

alabou commented on docs/Controllers.md:
```diff
@@ -65,7 +65,7 @@ Controllers MUST observe and interpret all of the TXT records returned with the
 ## Query API
 
-The Controller MUST be capable of using the Query API to discover any registered resource, including Node, Device, Source, Flow, Sender, and Receiver,
+In registered operation, the Controller MUST use the Query API to discover any registered resource, including Node, Device, Source, Flow, Sender, and Receiver,
```
Is there a definition of "registered operation" ?

N-Nagorny replied:
It's covered in `Discovery` section. It would be good to capitalize "Registered Operation" everywhere in this PR.

garethsb replied:
Done, thanks!

garethsb committed:
Clarify references to Registered Operation and Peer-to-Peer Operation

garethsb committed:
Tweaking wording of Query API: Client Interaction Procedure...
Changing as little as possible, use client as general term for Node or, much more likely when thinking about Query API discovery, a "control interface", i.e. Controller.

peterbrightwell reviewed (approved):
Looks to improve language
